compiler/src/dotty/tools/dotc/core/Flags.scala

```diff
@@ -172,12 +172,24 @@ object Flags {
   }
 
   /** The union of all flags in given flag set */
-  def union(flagss: FlagSet*) = (EmptyFlags /: flagss)(_ | _)
+  def union(flagss: FlagSet*): FlagSet = {
```

Comment:
This is still going to box, isn't it? `FlagSet*` becomes `Seq[FlagSet]`, value classes in generic positions are always boxed.

Reply:
We avoid reboxing in the fold, half of the boxes are gone.